docs/digital_analog_qc/daqc-cnot.md

```diff
@@ -43,160 +43,166 @@ $$\begin{aligned}
 
 ```python exec="on" source="material-block" result="json" session="daqc-cnot"
 # CNOT with Hamiltonian Evolution
-cnot_evo = qd.chain(
+cnot_evo = chain(
     H(1),
     cphase_evo,
     H(1)
 )
 
-init_state = qd.product_state("10")
+# Initialize state to check CNOTs sample outcomes.
+init_state = product_state("10")
 
-print(qd.sample(n_qubits, block = cnot_gate, state = init_state, n_shots = 100))
-print(qd.sample(n_qubits, block = cnot_evo, state = init_state, n_shots = 100))
+print(f"sample cnot_gate = {sample(n_qubits, block = cnot_gate, state = init_state, n_shots = 100)}") # markdown-exec: hide
+print(f"sample cnot_evo = {sample(n_qubits, block = cnot_evo, state = init_state, n_shots = 100)}") # markdown-exec: hide
 ```
 
-Thus, a CNOT gate can be applied by combining a few single-qubit gates together with a 2-qubit Ising interaction between the control and the target qubit. This is important because it now allows us to exemplify the usage of the Ising transform proposed in the DAQC paper [^2]. In the paper, the transform is described for $ZZ$ interactions. In `qadence` it works both with $ZZ$ and $NN$ interactions.
+Thus, a `CNOT` gate can be created by combining a few single-qubit gates together with a two-qubit Ising interaction between the control and the target qubit which is the essence of the Ising transform proposed in the seminal DAQC paper[^2] for $ZZ$ interactions. In Qadence, both $ZZ$ and $NN$ interactions are supported.
 
-## CNOT in an interacting system of 3 qubits
+## `CNOT` in an interacting system of three qubits
 
-Consider a simple experimental setup with $n=3$ interacting qubits in a triangular grid. For simplicity let's consider that all qubits interact with each other with an Ising ($NN$) interaction of constant strength $g_\text{int}$. The Hamiltonian for the system can be written by summing this interaction over all pairs:
+Consider a simple experimental setup with $n=3$ interacting qubits laid out in a triangular grid. For the sake of simplicity, all qubits interact with each other with an $NN$-Ising interaction of constant strength $g_\text{int}$. The Hamiltonian for the system can be written by summing interaction terms over all pairs:
 
 $$\mathcal{H}_\text{sys}=\sum_{i=0}^{n}\sum_{j=0}^{i-1}g_\text{int}N_iN_j,$$
 
 which in this case leads to only three interaction terms,
 
 $$\mathcal{H}_\text{sys}=g_\text{int}(N_0N_1+N_1N_2+N_0N_2)$$
 
-This generator can be easily built:
-
+This generator can be easily built in Qadence:
 
 ```python exec="on" source="material-block" result="json" session="daqc-cnot"
+from qadence import add, kron
 n_qubits = 3
 
+# Interaction strength.
 g_int = 1.0
 
+# Build a list of interactions.
 interaction_list = []
 for i in range(n_qubits):
     for j in range(i):
-        interaction_list.append(g_int * qd.kron(N(i), N(j)))
+        interaction_list.append(g_int * kron(N(i), N(j)))
 
-h_sys = qd.add(*interaction_list)
+h_sys = add(*interaction_list)
```

Comment:
Now this can also just be:

```
from qadence import hamiltonian_factory, Interaction

n_qubits = 3

h_sys = hamiltonian_factory(n_qubits, interaction = Interaction.NN)
```

Reply:
Thanks a lot ! Typical João style to add spaces around operators in arguments :)